osu.Game/Screens/Edit/EditorChangeHandler.cs

```diff
@@ -34,7 +36,8 @@ public class EditorChangeHandler : IEditorChangeHandler
         /// Creates a new <see cref="EditorChangeHandler"/>.
         /// </summary>
         /// <param name="editorBeatmap">The <see cref="EditorBeatmap"/> to track the <see cref="HitObject"/>s of.</param>
-        public EditorChangeHandler(EditorBeatmap editorBeatmap)
+        /// <param name="beatmapSkin">The skin to track the inline skin configuration of.</param>
+        public EditorChangeHandler(EditorBeatmap editorBeatmap, LegacyBeatmapSkin beatmapSkin)
```

Comment:
There won't be any issues, but by structure, `EditorBeatmap` is a wrapper for `IBeatmap` and is tracking any changes to the hit objects list, so for me that doesn't sound like the beatmap skin should be part of it, unless you plan for the `EditorBeatmap` to be more than just an `IBeatmap` wrapper.

Reply:
Also for not taking a `LegacyBeatmapSkin`, initially it was taking an `ISkin` for beatmap skins, but I was against it for the fact that it could take literally any skin, not only beatmap skins. https://github.com/ppy/osu/pull/9957#discussion_r475192958

I've proposed having an `IBeatmapSkin` interface and using it in all the beatmap skin code paths.

Reply:
The legacy skin constraint should apply only to the legacy decoder. Again, I'm not sure what an interface would achieve here.

Reply:
Yeah, I meant it to be a marker interface as a constraint to only skins created by `WorkingBeatmap`s, not any skin.

Reply:
Any guidance on how to implement a skin property inside the EditorBeatmap? Otherwise, I'll try figuring this out by myself.

Reply:
Just add it as a normal property, what's the problem?

Reply:
> Also for not taking a `LegacyBeatmapSkin`, initially it was taking an `ISkin` for beatmap skins, but I was against it for the fact that it could take literally any skin, not only beatmap skins. [#9957 (comment)](https://github.com/ppy/osu/pull/9957#discussion_r475192958)

@frenzibyte care to explain? Why should the beatmap _not_ be able to take in any skin?

Reply:
I've suggested it only as a constraint that `BeatmapManager.Save` for example only allows saving skins that come from `WorkingBeatmap`, but actually that sounds unnecessary... I take that back, and using `ISkin` in its place is correct, perhaps I rushed in thinking of suggesting it.